TCMSApp/src/server/model/mschemas/run.js

```diff
@@ -25,45 +25,10 @@ var Schema = mongoose.Schema;
         envFull: {},
         status: {
             type: String,
-            validate: {
-                validator: function (v) {
-                    return /passed|executing|failed/.test(v);
-                },
-                message: '{VALUE} is not valid status!'
-            },
-            default: 'executing'
+            enum: ['passed', 'executing', 'failed', 'none'],
+            default: 'none'
         },
-        //the
-        tests: [{
-            testName: {type: String, required: true},
-            testDescription: {type: String, required: true},
-            automated: Boolean,
-            preConditions: {type: String, required: true},
-            steps: [{
-                stepNumber: Number,
-                stepDescription: {type: String, required: true},
-                expectedResult: {type: String, required: true},
-
-                status: {
-                    type: String,
-                    validate: {
-                        validator: function (v) {
-                            return /passed, blocked, failed/.test(v);
-                        },
-                        message: '{VALUE} is not valid status!'
-                    }
-                }
-            }],
-            status: {
-                type: String,
-                validate: {
-                    validator: function (v) {
-                        return /passed|executing|failed/.test(v);
-                    },
-                    message: '{VALUE} is not valid status!'
-                }
-            }
-        }]
+        tests: [mongoose.Schema.Types.ObjectId]
```

Comment:
I think we should leave the tests here.
why?
1. we need to store information on failing/passing steps.
2. test case can be changed (but we need to store in the test run the state of test case for the moment of the test run!)